src/neo/SmartContract/Native/ContractManagement.cs

```diff
@@ -229,7 +229,7 @@ private void Update(ApplicationEngine engine, byte[] nefFile, byte[] manifest, S
             engine.SendNotification(Hash, "Update", new VM.Types.Array { contract.Hash.ToArray() });
         }
 
-        [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.WriteStates | CallFlags.AllowNotify)]
+        [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.States | CallFlags.AllowNotify)]
```

Comment:
I think we don't need `ReadStates` for `Destroy`. Because it only deletes data, not reads.

Reply:
First read the contract in order to check if exists